transformer/src/main/scala/uk/ac/wellcome/transformer/modules/StreamRecordProcessorFactoryModule.scala

```diff
@@ -37,21 +34,20 @@ class StreamsRecordProcessor(
     records: JList[Record],
     checkpointer: IRecordProcessorCheckpointer
   ): Unit = {
-    records.asScala.map { record =>
-      reciever.map(_ ! record)
+    records.asScala.foreach { record =>
+      receiver.receiveRecord(record)
```

Comment:
```scala
records.asScala.map(receiver.receiveRecord)
```

Would that work? I think that’s equivalent and looks a little cleaner, but there may well be a subtlety I’m missing.